trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/utilisation-details-helper.test.ts

```diff
@@ -0,0 +1,225 @@
+      expect(utilisationTableRow.exporter).toEqual(feeRecordUtilisation.exporter);
+    });
+
+    it('should map the base currency', () => {
+      // Arrange
+      const feeRecordUtilisation: FeeRecordUtilisation = {
+        ...aFeeRecordUtilisation(),
+        baseCurrency: CURRENCY.USD,
+      };
+
+      // Act
+      const utilisationTableRow = mapToUtilisationTableRowViewModel(feeRecordUtilisation, aDataSortValues());
+
+      // Assert
+      expect(utilisationTableRow.baseCurrency).toEqual(feeRecordUtilisation.baseCurrency);
+    });
+
+    it('should format the value with thousands separators and decimal places', () => {
+      // Arrange
+      const feeRecordUtilisation: FeeRecordUtilisation = {
+        ...aFeeRecordUtilisation(),
+        value: 456000,
+      };
+
+      // Act
+      const utilisationTableRow = mapToUtilisationTableRowViewModel(feeRecordUtilisation, aDataSortValues());
+
+      // Assert
+      expect(utilisationTableRow.formattedValue).toEqual('456,000.00');
+    });
+
+    it('should format the utilisation with thousands separators and decimal places', () => {
+      // Arrange
+      const feeRecordUtilisation: FeeRecordUtilisation = {
+        ...aFeeRecordUtilisation(),
+        utilisation: 123000,
+      };
+
+      // Act
+      const utilisationTableRow = mapToUtilisationTableRowViewModel(feeRecordUtilisation, aDataSortValues());
+
+      // Assert
+      expect(utilisationTableRow.formattedUtilisation).toEqual('123,000.00');
+    });
+
+    it('should map the cover percentage', () => {
+      // Arrange
+      const feeRecordUtilisation: FeeRecordUtilisation = {
+        ...aFeeRecordUtilisation(),
+        coverPercentage: 80,
+      };
+
+      // Act
+      const utilisationTableRow = mapToUtilisationTableRowViewModel(feeRecordUtilisation, aDataSortValues());
+
+      // Assert
+      expect(utilisationTableRow.coverPercentage).toEqual(feeRecordUtilisation.coverPercentage);
+    });
+
+    it('should format the exposure with thousands separators and decimal places', () => {
+      // Arrange
+      const feeRecordUtilisation: FeeRecordUtilisation = {
+        ...aFeeRecordUtilisation(),
+        exposure: 80000,
+      };
+
+      // Act
+      const utilisationTableRow = mapToUtilisationTableRowViewModel(feeRecordUtilisation, aDataSortValues());
+
+      // Assert
+      expect(utilisationTableRow.formattedExposure).toEqual('80,000.00');
+    });
+  });
+
+  describe('mapToUtilisationDetailsViewModel', () => {
+    it('maps each item to a utilisation table row', () => {
+      // Arrange
+      const items: FeeRecordUtilisation[] = [
+        { ...aFeeRecordUtilisation(), feeRecordId: 1 },
+        { ...aFeeRecordUtilisation(), feeRecordId: 2 },
+      ];
+
+      // Act
+      const viewModel = mapToUtilisationDetailsViewModel(items);
+
+      // Assert
+      expect(viewModel.utilisationTableRows.length).toEqual(2);
+      expect(viewModel.utilisationTableRows[0].feeRecordId).toEqual(1);
```

Comment:
Can we test the whole result instead please?

Reply:
Not easily no, since the mapping of individual records is covered above comprehensively it would be very a duplicated effort to figure out what an entire record should map to here, when actually all this test cares about is each one does get mapped once

Reply:
OK great thanks, as long as we have an instance of covering the whole result above like you say 👍